### PR TITLE
Update Tantivy to 0.8

### PR DESCRIPTION
**Pull request by osyoyu**
This pull request updates Tantivy to 0.8.x.

I've basically done:
- Update `tantivy` to 0.8 in Cargo.toml && `cargo update`
- Modify some code to follow the new Collector API in tantivy 0.8
- and resolve some Rust warnings

### CODE REVIEW

**Comment by osyoyu** [comment]
@fulmicoton Can you take a look at this?

File: src/commands/search.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ fn run_search(directory: &Path, query: &str) -> tantivy::Result<()> {
         let mut scorer = weight.scorer(segment_reader)?;
         while scorer.advance() {
             let doc_id = scorer.doc();
-            let doc = segment_reader.doc(doc_id)?;
+            let doc = segment_reader.get_store_reader().get(doc_id)?;
```
Comment (fulmicoton) [inline src/commands/search.rs]: can you get the store reader outside of the loop?

File: src/commands/serve.rs [hunk 2]
```diff
@@ -119,29 +119,25 @@ impl IndexServer {
...
         let mut timer_tree = TimerTree::default();
-        {
-            let _search_timer = timer_tree.open("search");
-            let mut chained_collector = collector::chain()
-                .push(&mut top_collector)
-                .push(&mut count_collector);
-            query.search(&searcher, &mut chained_collector)?;
-        }
+        let (doc_count, top_docs): (usize, Vec<(Score, DocAddress)>) = {
+            let mut _search_timer = timer_tree.open("search");
```
Comment (fulmicoton) [inline src/commands/serve.rs]: why mut?

File: src/commands/search.rs (hunk 1 above)
Comment (osyoyu) [reply src/commands/search.rs]: Sure! https://github.com/tantivy-search/tantivy-cli/pull/37/commits/7720853d4a37aec14411f49d9f7c283b34ee765f

File: src/commands/serve.rs (hunk 2 above)
Comment (osyoyu) [reply src/commands/serve.rs]: Got in while fiddling around; removed

**Comment by fulmicoton** [comment]
oops sorry I should have merged that earlier. Jumped version to 0.9.

**Pull request closed without merging**